src/hotspot/share/gc/parallel/parallelScavengeHeap.hpp

```diff
@@ -286,4 +290,19 @@ class AdaptiveSizePolicyOutput : AllStatic {
   }
 };
 
+// The HeapBlockClaimer is used during parallel iteration over heap,
+// allowing workers to claim heap blocks, gaining exclusive rights to these blocks.
+// The eden and survivor spaces are treated as single blocks as it is hard to divide
+// these spaces.
+// The old spaces are divided into serveral fixed-size blocks.
+class HeapBlockClaimer : public StackObj {
```

Comment:
This class could probably be moved to the cpp file. It seems only be used in the parallelScavengeHeap.cpp file.